### PR TITLE
Stronger SAML attribute linking

### PR DESCRIPTION
**Pull request by andreaceccanti**
This PR provides a fix for issue #116, introducing 

- a new attributeId column in the IAM_SAML_ID table which holdes the name of the SAML attribute used for the linking
- tests to check that linking works as expected
- changes to the dashboard to include the new attribute id

### CODE REVIEW

File: iam-login-service/src/main/java/it/infn/mw/iam/config/JpaConfig.java [hunk 1]
```diff
@@ -43,9 +43,9 @@ protected AbstractJpaVendorAdapter createJpaVendorAdapter() {
     map.put("eclipselink.logging.level.sql", "OFF");
     map.put("eclipselink.cache.shared.default", "false");
 
-    // map.put("eclipselink.ddl-generation.output-mode", "sql-script");
-    // map.put("eclipselink.ddl-generation", "create-tables");
-    // map.put("eclipselink.create-ddl-jdbc-file-name", "ddl.sql");
+    map.put("eclipselink.ddl-generation.output-mode", "sql-script");
+    map.put("eclipselink.ddl-generation", "create-tables");
+    map.put("eclipselink.create-ddl-jdbc-file-name", "ddl.sql");
```
Comment (enricovianello) [inline iam-login-service/src/main/java/it/infn/mw/iam/config/JpaConfig.java]: Should be commented again

File: iam-login-service/src/main/webapp/resources/iam/js/dashboard-app/dashboard-app.module.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ angular.module('dashboardApp')
       $rootScope.iamVersion = getIamVersion();
       $rootScope.iamCommitId = getIamGitCommitId();
       
-      LoadTemplatesService.loadTemplates();
+      // LoadTemplatesService.loadTemplates();
```
Comment (enricovianello) [inline iam-login-service/src/main/webapp/resources/iam/js/dashboard-app/dashboard-app.module.js]: Remove comment

File: iam-persistence/src/main/resources/db/migration/h2/V13__add_attribute_id_to_saml_id_table.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+-- Add attribute_id column
+ALTER TABLE iam_saml_id ADD COLUMN (attribute_id varchar(256));
+-- Create index
+CREATE INDEX IDX_IAM_SAML_ID_1 ON iam_saml_id(IDPID, attribute_id, USERID);
+-- Update existing records, so that epuid is the default attribute id
+update iam_saml_id set attribute_id = 'urn:oid:1.3.6.1.4.1.5923.1.1.1.13' where attribute_id = NULL;
```
Comment (marcocaberletti) [inline iam-persistence/src/main/resources/db/migration/h2/V13__add_attribute_id_to_saml_id_table.sql]: The condition "attribute_id = NULL" is always false.
The correct statement is "where attribute_id IS NULL"

File: iam-persistence/src/main/resources/db/migration/mysql/V13__add_attribute_id_to_saml_id_table.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+-- Add attribute_id column
+ALTER TABLE iam_saml_id ADD COLUMN (attribute_id varchar(256));
+-- Create index
+CREATE INDEX IDX_IAM_SAML_ID_1 ON iam_saml_id(IDPID, attribute_id, USERID);
+-- Update existing records, so that epuid is the default attribute id
+update iam_saml_id set attribute_id = 'urn:oid:1.3.6.1.4.1.5923.1.1.1.13' where attribute_id = NULL;
```
Comment (marcocaberletti) [inline iam-persistence/src/main/resources/db/migration/mysql/V13__add_attribute_id_to_saml_id_table.sql]: Same comment as above

**Pull request merged**